### DISCUSSION TITLE
Codespace Build Fails With: `error: failed to receive status: rpc error: code = Unavailable desc = error reading from server: EOF`

### DISCUSSION THREAD

**Discussion by bgianfo** [community/community#53271] [Codespaces] [answered]
### Select Topic Area

Bug

### Body

My codespaces devcontainer build is consistently failing with the following cryptic message: `error: failed to receive status: rpc error: code = Unavailable desc = error reading from server: EOF`

The following issues mention the same error and it seems to point back to a bug in docker (https://github.com/moby/moby/issues/44918): 
- https://github.com/microsoft/vscode-remote-release/issues/6814
- https://github.com/microsoft/vscode-remote-release/issues/7958#issuecomment-1416495001

The bug is reportedly fixed in the latest version of docker, released in February. However I continue to see what appears to be the same bug in the codespace container builds for my private repo. From the logs I'm unable to see which version of docker is being used to build the codespace devcontainer. 

I've also attempted to use the workaround I've seen mentioned various places, of attempting to set `BUILDKIT_INLINE_CACHE=0`. But this doesn't seem to actually work around the issue I am seeing. 

```json
    "build": {
        "dockerfile": "Dockerfile",
        "args": {
          "BUILDKIT_INLINE_CACHE": "0"
        }
    },

```

I wanted to report this issue, to see if it's already known, or if there are any other known workarounds possible? It would also be nice if the codespace build emitted some version information. Any help would be greatly appreciated. 

----
The entire error message below:
```
#24 exporting to image
#24 exporting layers
error: failed to receive status: rpc error: code = Unavailable desc = error reading from server: EOF
Stop: Run: docker buildx build --load --build-arg BUILDKIT_INLINE_CACHE=1 -f /tmp/devcontainercli-root/container-features/0.37.0-1681944348243/Dockerfile-with-features -t vsc-apps-27a3563183f41e8de9c5b20ed5108d9696cbef2345000a3ced00c0f02890d6fd --target dev_containers_target_stage --build-arg BUILDKIT_INLINE_CACHE=0 --build-arg _DEV_CONTAINERS_BASE_IMAGE=dev_container_auto_added_stage_label --build-arg _DEV_CONTAINERS_IMAGE_USER=root --build-arg _DEV_CONTAINERS_FEATURE_CONTENT_SOURCE=dev_container_feature_content_temp /var/lib/docker/codespacemount/workspace/apps/.devcontainer
Error: Command failed: docker buildx build --load --build-arg BUILDKIT_INLINE_CACHE=1 -f /tmp/devcontainercli-root/container-features/0.37.0-1681944348243/Dockerfile-with-features -t vsc-apps-27a3563183f41e8de9c5b20ed5108d9696cbef2345000a3ced00c0f02890d6fd --target dev_containers_target_stage --build-arg BUILDKIT_INLINE_CACHE=0 --build-arg _DEV_CONTAINERS_BASE_IMAGE=dev_container_auto_added_stage_label --build-arg _DEV_CONTAINERS_IMAGE_USER=root --build-arg _DEV_CONTAINERS_FEATURE_CONTENT_SOURCE=dev_container_feature_content_temp /var/lib/docker/codespacemount/workspace/apps/.devcontainer
    at _se (/usr/lib/node_modules/@devcontainers/cli/dist/spec-node/devContainersSpecCLI.js:1917:1698)
    at runMicrotasks (<anonymous>)
    at processTicksAndRejections (internal/process/task_queues.js:95:5)
    at async CD (/usr/lib/node_modules/@devcontainers/cli/dist/spec-node/devContainersSpecCLI.js:1916:3889)
    at async K7 (/usr/lib/node_modules/@devcontainers/cli/dist/spec-node/devContainersSpecCLI.js:1916:2808)
    at async Hse (/usr/lib/node_modules/@devcontainers/cli/dist/spec-node/devContainersSpecCLI.js:1931:2626)
    at async Ch (/usr/lib/node_modules/@devcontainers/cli/dist/spec-node/devContainersSpecCLI.js:1931:3741)
    at async cae (/usr/lib/node_modules/@devcontainers/cli/dist/spec-node/devContainersSpecCLI.js:2059:17376)
    at async uae (/usr/lib/node_modules/@devcontainers/cli/dist/spec-node/devContainersSpecCLI.js:2059:17117)
{"outcome":"error","message":"Command failed: docker buildx build --load --build-arg BUILDKIT_INLINE_CACHE=1 -f /tmp/devcontainercli-root/container-features/0.37.0-1681944348243/Dockerfile-with-features -t vsc-apps-27a3563183f41e8de9c5b20ed5108d9696cbef2345000a3ced00c0f02890d6fd --target dev_containers_target_stage --build-arg BUILDKIT_INLINE_CACHE=0 --build-arg _DEV_CONTAINERS_BASE_IMAGE=dev_container_auto_added_stage_label --build-arg _DEV_CONTAINERS_IMAGE_USER=root --build-arg _DEV_CONTAINERS_FEATURE_CONTENT_SOURCE=dev_container_feature_content_temp /var/lib/docker/codespacemount/workspace/embedded-apps/.devcontainer","description":"An error occurred building the image."}
devcontainer process exited with exit code 1
Failed to create container.
Error: Command failed: docker buildx build --load --build-arg BUILDKIT_INLINE_CACHE=1 -f /tmp/devcontainercli-root/container-features/0.37.0-1681944348243/Dockerfile-with-features -t vsc-apps-27a3563183f41e8de9c5b20ed5108d9696cbef2345000a3ced00c0f02890d6fd --target dev_containers_target_stage --build-arg BUILDKIT_INLINE_CACHE=0 --build-arg _DEV_CONTAINERS_BASE_IMAGE=dev_container_auto_added_stage_label --build-arg _DEV_CONTAINERS_IMAGE_USER=root --build-arg _DEV_CONTAINERS_FEATURE_CONTENT_SOURCE=dev_container_feature_content_temp /var/lib/docker/codespacemount/workspace/apps/.devcontainer
Error Code: 1302
Container creation failed.
Jobs failed, exiting the agent. Error Code: 1302
Error: Process completed with exit code 1.
```

**Comment by eljog** [comment]
Thank you for reporting this issue. I have filed an internal issue with our engineering team to investigate.

**Comment by eljog** [comment, marked as answer]
While we look into a permanent fix, would you like to try the following workaround?

I understand the workaround to set `"BUILDKIT_INLINE_CACHE": "0"` did not work. Along with that workaround it might worth trying to disable BuildKit. One way to disable build kit today is by [adding a secret](https://github.com/settings/codespaces/secrets/new) in GitHub named  `DOCKER_BUILDKIT` with value `0`

**Reply by bgianfo** [reply]
Thanks, that workaround seems to have worked in my environment. I set a codespace secret with `DOCKER_BUILDKIT` set to `0` in my repository, and the build seems to be consistently working.

Thanks!

**Comment by oliver-platt** [comment]
Hey, I'm also having this issue but neither setting `"BUILDKIT_INLINE_CACHE": "0"` in the `args` nor the repository Codespaces secret seems to have worked for me unfortunately. Has there been any progress with this issue?

**Comment by oliver-platt** [comment]
I've ran a few tests, and I'm able to get the Codespace to build by removing all features (e.g. docker in docker, ZSH, etc). Seems that adding any of these back in causes the build to fail, interestingly with an error that seems to show the `BUILDKIT_INLINE_CACHE=1` despite being set to `0` in the args. 

Unfortunately, removing these features makes Codespaces useless to me - I need docker in docker in particular to create my environment.

Would appreciate any insight here as to anything I can do

**Comment by benkeen** [comment]
We're encountering the same problem. Neither the `BUILDKIT_INLINE_CACHE` suggestion nor the  accepted solution of the `DOCKER_BUILDKIT` secret worked for us. I tried two different ubuntu 22-based images and get the same error on both.

Are there any other options available to get around this issue?
